lib/ftp/src/ftp.erl

```diff
@@ -75,39 +75,41 @@
 
 %% Internal state
 -record(state, {
-          csock   = undefined, % socket() - Control connection socket
-          dsock   = undefined, % socket() - Data connection socket
-          tls_options = undefined, % list()
-          verbose = false, % boolean()
-          ldir    = undefined, % string() - Current local directory
-          type    = ftp_server_default, % atom() - binary | ascii
-          chunk   = false, % boolean() - Receiving data chunks
-          mode    = ?DEFAULT_MODE, % passive | active
-          timeout = ?CONNECTION_TIMEOUT, % integer()
+          csock   = undefined :: undefined  % Control connection socket
+                               | { tcp | ssl, inet:socket() | ssl:socket() | ssl:sslsocket()},
+          dsock   = undefined :: undefined % Data connection socket
+                               | { tcp | ssl | lsock, inet:socket() | ssl:socket() | ssl:sslsocket() | gen_tcp:socket()},
+          tls_options = undefined :: undefined | [tuple()],
+          verbose = false  :: boolean(),
+          ldir    = undefined :: undefined | file:filename_all(), % Current local directory
+          type    = ftp_server_default :: atom(), % binary | ascii
+          chunk   = false :: boolean(), % Receiving data chunks
+          mode    = ?DEFAULT_MODE :: active | passive,
+          timeout = ?CONNECTION_TIMEOUT :: integer(),
```

Comment:
I think it should be timeout as it is passed to gen_tcp:connect that expects a timeout()